simulator/func_sim/alu.h

```diff
@@ -203,7 +203,17 @@ struct ALU
     // Leading zero/ones
     template<typename I, typename T> static void clo( I* instr)  { instr->v_dst = count_leading_ones<T>( instr->v_src1); }
     template<typename I, typename T> static void clz( I* instr)  { instr->v_dst = count_leading_zeroes<T>( instr->v_src1); }
-
+    template<typename I, typename T> static
+    void pcnt(I* instr){
+      std::size_t max = bitwidth<T> - 1 - count_leading_zeroes<T>( ~instr->v_src1);
+      uint8 count = 0;
+      for ( std::size_t index = 0; index < max; index++)
+      {
+        count += (instr->v_src1 >> index) & 1;
+      }
+      instr->v_dst = count;
```

Comment:
Have you tried to use this function?

https://github.com/MIPT-ILab/mipt-mips/blob/45744d4055414a0982fe755a6ae88ea7279417dd/simulator/infra/macro.h#L53